### PR TITLE
Feat/favoriting

### PR DESCRIPTION
**Pull request by isleofyou**
What (if any) features are you implementing?
- Added favoriting feature that finds the item in state, updates the favorite property, and sets state again.

Were there any issues that arose?
- Got stuck trying to use prevState which seemed the right move but had to remove it for now. Would be interesting to see this implemented. 

Any other comments, questions, or concerns?
Huge thanks to Markus for doing a lot of this. also thanks to Dustin (Markus' friend) for doing some troubleshooting.

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ import Header from '../Header/Header';
 import { Component } from 'react';
 import Aside from '../Aside/Aside';
 import ProductsContainer from '../ProductsContainer/ProductsContainer';
-import Loading from '../Loading/Loading';
 import { BrowserRouter as Router, Routes, Route} from 'react-router-dom';
-import { getAllProducts } from '../../apiCalls';
+import { getAllProducts, updateFavorite } from '../../apiCalls';
+//import Error component here
```
Comment (ivonne-hernandez) [inline src/Components/App/App.js]: Once we get the Error component merged into main, we'll want to import it here and remove this comment.

**Review by ivonne-hernandez — APPROVED** [review]
This is ready to be merged with main.

File: src/Components/App/App.js [hunk 2]
```diff
@@ -21,36 +20,47 @@ class App extends Component {
...
-    if (this.state.products.length > 0) {
-      return (
-        <ProductsContainer products={this.state.products} />
-      )
-    } else {
-      return (
-        <Loading />
-      )
-    }
+  addFavorite = (id) => {
```
Comment (Markus-Xavier) [inline src/Components/App/App.js:32]: I like how this is being done on, I think this is a great solution

**Review by Markus-Xavier — APPROVED** [review]
This is ready to be merged into main

**Pull request merged**